bugsnag-android-core/src/main/java/com/bugsnag/android/EventStore.java

```diff
@@ -58,64 +60,78 @@ public int compare(File lhs, File rhs) {
-
-            if (!crashReports.isEmpty()) {
-
-                // Block the main thread for a 2 second interval as the app may crash very soon.
-                // The request itself will run in a background thread and will continue after the 2
-                // second period until the request completes, or the app crashes.
-                flushOnLaunchCompleted = false;
-                logger.i("Attempting to send launch crash reports");
-
-                try {
-                    backgroundTaskService.submitTask(TaskType.ERROR_REQUEST, new Runnable() {
-                        @Override
-                        public void run() {
-                            flushReports(crashReports);
-                            flushOnLaunchCompleted = true;
-                        }
-                    });
-                } catch (RejectedExecutionException ex) {
-                    logger.w("Failed to flush launch crash reports", ex);
-                    flushOnLaunchCompleted = true;
+        if (!config.getSendLaunchCrashesSynchronously()) {
+            return;
+        }
+        Future<?> future = null;
+        try {
+            future = bgTaskSevice.submitTask(TaskType.ERROR_REQUEST, new Runnable() {
+                @Override
+                public void run() {
+                    flushLaunchCrashReport();
                 }
+            });
+        } catch (RejectedExecutionException exc) {
+            logger.d("Failed to flush launch crash reports, continuing.", exc);
+        }
+
+        try {
+            if (future != null) {
+                future.get(LAUNCH_CRASH_TIMEOUT_MS, TimeUnit.MILLISECONDS);
+            }
+        } catch (InterruptedException | ExecutionException | TimeoutException exc) {
+            logger.d("Failed to send launch crash reports within 2s timeout, continuing.", exc);
+        }
+    }
 
-                long waitMs = 0;
+    void flushLaunchCrashReport() {
+        List<File> storedFiles = findStoredFiles();
+        File launchCrashReport = findLaunchCrashReport(storedFiles);
 
-                while (!flushOnLaunchCompleted && waitMs < LAUNCH_CRASH_TIMEOUT_MS) {
-                    try {
-                        Thread.sleep(LAUNCH_CRASH_POLL_MS);
-                        waitMs += LAUNCH_CRASH_POLL_MS;
-                    } catch (InterruptedException exception) {
-                        logger.w("Interrupted while waiting for launch crash report request");
-                    }
-                }
-                logger.i("Continuing with Bugsnag initialisation");
-            } else {
-                logger.d("No startupcrash events to flush to Bugsnag.");
+        // cancel non-launch crash reports
+        if (launchCrashReport != null) {
+            storedFiles.remove(launchCrashReport);
+        }
+        cancelQueuedFiles(storedFiles);
+
+        if (launchCrashReport != null) {
+            logger.i("Attempting to send launch crash reports");
```

Comment:
This message should reflect the new intent of sending only one crash report. This change should avoid any confusion around what happens when there is more than one on disk.

```suggestion
            logger.i("Attempting to send the most recent launch crash report");
```